src/Utilities/QuaternionHelpers.hpp

```diff
@@ -0,0 +1,39 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines helper functions for converting between `std::array` and boost
+/// quaternions.
+
+#pragma once
+
+#include <array>
+#include <boost/math/quaternion.hpp>
+
+#include "Utilities/Gsl.hpp"
+
+template <typename T>
+std::array<T, 4> quaternion_to_array(
```

Comment:
Please add some documentation to these functions, including descriptions of the argument and the return value.